src/dataProvider/model/MediaComposition.js

```diff
@@ -0,0 +1,322 @@
+      dvr: resource.dvr,
+      eventData: this.getMainChapter().eventData,
+      id: this.getMainChapter().id,
+      imageCopyright: this.getMainChapter().imageCopyright,
+      live: resource.live,
+      mediaType: this.getMainChapter().mediaType,
+      mimeType: resource.mimeType,
+      presentation: resource.presentation,
+      quality: resource.quality,
+      streaming: resource.streaming,
+      streamOffset: resource.streamOffset,
+      tokenType: resource.tokenType,
+      url: resource.url,
+    }));
+  }
+
+  /**
+   * Get segments of the main chapter ordered by markIn.
+   *
+   * @returns {Array} of segments
+   */
+  getMainSegments() {
+    const mainChapter = this.getMainChapter();
+
+    if (!this.mainSegments && mainChapter && mainChapter.segmentList) {
+      this.mainSegments = mainChapter.segmentList;
+    }
+
+    return this.mainSegments || [];
+  }
+
+  /**
+   * Compute a date from which this content is valid. Always return a date object.
+   *
+   * @returns {Date} date specified in media composition or EPOCH when no date present.
+   */
+  getMainValidFromDate() {
+    const mainChapter = this.getMainChapter();
+
+    if (!mainChapter) {
+      return new Date(0);
+    }
+
+    const { validFrom } = mainChapter;
+
+    if (validFrom) {
+      return new Date(validFrom);
+    }
+  }
+
+  /**
+   * Get merged analytics data.
+   *
+   * @returns {Object}
+   */
+  getMergedAnalyticsData(analyticsData) {
+    return {
+      ...this.analyticsData,
+      ...this.getMainChapter().analyticsData,
+      ...analyticsData,
+    };
+  }
+
+  /**
+   * Get merged analytics metadata.
+   *
+   * @returns {Object}
+   */
+  getMergedAnalyticsMetadata(analyticsMetadata) {
+    return {
+      ...this.analyticsMetadata,
+      ...this.getMainChapter().analyticsMetadata,
+      ...analyticsMetadata,
+    };
+  }
+
+  /**
+   * Get the chapter's resource list
+   * @returns {Array} of resources
+   */
+  getResourceList() {
+    const { resourceList } = this.getMainChapter();
+
+    return resourceList || [];
+  }
+
+  /**
+   * Get subdivisions to be displayed for this MediaComposition.
+   * They can be segments or chapters.
+   *
+   * @returns {Array} of subdivisions
+   */
+  getSubdivisions() {
+    const chapters = this.getChapters();
+    const mainChapter = this.getMainChapter();
+    const subdivisions = [];
+    const displayableMainSegments = this.getMainSegments().filter(
+      (s) => s.displayable
+    );
+    const hasDisplayableSegments = Boolean(displayableMainSegments.length);
+
+    if (chapters.length === 1 && !hasDisplayableSegments) {
+      return [];
+    }
+
+    chapters.forEach((chapter) => {
+      const isSegment =
+        chapter.urn === mainChapter.urn && hasDisplayableSegments;
+
+      if (isSegment) {
+        subdivisions.push(...displayableMainSegments);
+      } else {
+        subdivisions.push(chapter);
+      }
+    });
+
+    return subdivisions;
+  }
+}
```

Comment:
Do we really need to parse segments at this stage (purpose of the PR is to find a chapter to play, isn't it)?

Reply:
No, that's clearly an oversight 👌🏽